### DISCUSSION TITLE
When does the view cache clear?

### DISCUSSION THREAD

**Discussion by ronen-e** [mobxjs/mobx-state-tree#1730] [Q&A] [answered]
Hi Everyone.

I was working with mobx-state-tree and I created a view for a simple fullName property combining firstName and lastName

seeing as how mobx-state-tree caches the results, I was wondering how can I be 100% certain the results would be computed again when needed (e.g when firstName changes).

I was hoping someone here knows or can point me in the right direction.

<img width="577" alt="Screen Shot 2021-06-24 at 1 32 21 PM" src="https://user-images.githubusercontent.com/10486251/123250052-51886700-d4f2-11eb-872e-3351ed56829d.png">

**Comment by jamonholmgren** [comment, marked as answer]
@ronen-e Any properties that are accessed within a view are then "remembered" as being accessed, which then ensures that the computation gets computed again when they change.

I don't have the code handy for showing you how this is done, but it seems to be pretty reliable to me.

**Reply by ronen-e** [reply]
@jamonholmgren I checked the source code (both `mobx-state-tree` and `mobx`) and it is reliable.
I noticed that when using `getters` they are wrapped with `mobx.makeObservable` and that when using functions they are assigned to the view model.

My specific case was using an impure function such as 
```js
getDate() {
  return self.name + ': ' + Date.now()
}
```
in this case the `Date.now()` makes it only useable as a view function in React components like this:
```jsx
<span>{store.getDate()}</span>
```

Not as a `getter`:
```js
get date() {
  return self.name + ': ' + Date.now()
}
```
```jsx
<span>{store.date}</span>
```
